### ISSUE TITLE
Add DetailedList to Windows

### ISSUE THREAD

**Issue by ibigpapa** [beeware/toga#502]
## Expected Behavior

The detailed list example to render

## Current Behavior

```python.exe -m detailedlist

Unhandled Exception: Python.Runtime.PythonException: AttributeError : module 'toga_winforms.factory' has no attribute 'DetailedList'
   at Python.Runtime.Dispatcher.Dispatch(ArrayList args)
   at __System_Threading_ThreadStartDispatcher.Invoke()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()

Process finished with exit code -532462766
```
## Steps to reproduce

1. launch detailedlist example

## Your Environment

* Python Version (list the specific version number) 3.6.5

* Operating System and Version (select from the following and list the specific version number; if your OS is not listed, list that as well)

  - [ ] macOS - version: 
  - [ ] Linux - distro:   - version:
  - [ x ] Windows - version: Win10 1607
  - [ ] Other - name:  - version:

* Toga Target (the type of app you are trying to generate)
 
  - [ ] android
  - [ ] cocoa
  - [ ] django 
  - [ ] gtk
  - [ ] iOS
  - [ ] tvOS
  - [ ] watchOS
  - [ x ] winforms 
  - [ ] win32
  - [ ] Other (please specify)

**Comment by SatyakiDe2019** [comment]
Guys!

I'm having a similar issue in Windows 10 with the Python 3.7.2.

`C:\Technical Docs\PythonCourse\Pandas\blog\Demo_31\SDChat>briefcase dev

[SDChat] Starting in dev mode...
Traceback (most recent call last):
  File "c:\python3_7_2\lib\site-packages\toga_winforms\app.py", line 128, in run_app
    self.create()
  File "c:\python3_7_2\lib\site-packages\toga_winforms\app.py", line 64, in create
    self.interface.startup()
  File "C:\Technical Docs\PythonCourse\Pandas\blog\Demo_31\SD\src\SD\app.py", line 21, in startup
    self.chat = toga.DetailedList(data=[])
  File "c:\python3_7_2\lib\site-packages\toga\widgets\detailedlist.py", line 35, in __init__
    self._impl = self.factory.DetailedList(interface=self)
AttributeError: module 'toga_winforms.factory' has no attribute 'DetailedList'`

Does anyone have any idea?

I tried to reinstall the latest dev version by uninstalling the previous stable version of toga.

Any suggestions will be highly appreciated. 

Regards.

**Comment by freakboy3742** [comment]
"toga_winforms.factory has no attribute DetaileList" is Toga's way of saying "This widget is not available on this platform yet". 

DetailedList exists on macOS, GTK and iOS, but hasn't been implemented on Windows.

**Comment by SatyakiDe2019** [comment]
Thanks for your confirmation. Is there any alternate way to achieve the same?

**Comment by freakboy3742** [comment]
Not really - if there was another way to build a DetailedList, we wouldn't need a DetailedList widget :-)

**Comment by SatyakiDe2019** [comment]
Okies. In that case, how are you going to build a chat application without that? Sorry if I'm asking some very basic questions. At this point, I'm trying to explore this framework.

**Comment by freakboy3742** [comment]
The short answer is that *right now*, you won't. 

Toga isn't a v1.0 library. It's a pre-release of v0.3. There are *many* features that aren't implemented. And the feature set available on some platforms (like Windows) are lagging behind what is possible on other platforms (like macOS). You're not going to be able to take Toga as-is today and build your commercial, cross-platform world-beating chat application. 

You will, however, be able to play around with some toy apps and see that a native cross platform widget toolkit is possible. Better still, we hope that what you see is enough to convince you to try your hand at contributing. One of the biggest problems we have is that we need people with platform knowledge and experience to help contribute. There's a lot that still needs to be done, and we need all the help we can get.

**Comment by SatyakiDe2019** [comment]
Thanks for your clarification. At this moment, I'm planning to post how one can use a beeware framework using python for various platform mobile app in my blog. Generally, I always post something which covers Windows & Mac/Ubuntu combinations. However, this time I might go with the Mac and/or Ubuntu combination.

**Comment by obulat** [comment]
> "toga_winforms.factory has no attribute DetailedList" is Toga's way of saying "This widget is not available on this platform yet".
Should we rewrite the error message? I think the second option would be easier for users to understand and would decrease the number of such issues. Maybe, during the active alpha stage we could add something like: "If you want to implement the widget for <platform>, <document> will help you get started" to encourage contributions?

**Comment by freakboy3742** [comment]
@obulat Agreed - that would be a nice improvement to the end-user experience. 

The problem here is that what we'd need to catch is an attribute lookup on a module. This is something that is trivial to fix in Python 3.7 - we could add a module-level `__getattr__` definition. However, on older Python versions, we'd need to completely change the way the factory is accessed (and prevent any other way of accessing the factory).

It may still be worth adding the module-level `__getattr__`, though - it won't *hurt* having it on older Python versions, and the platform most likely to be affected by this (Windows) is also the one most likely to have a recent Python version (3.7+).

**Comment by saroad2** [comment]
Reviving this issue.

Do we have any idea on how to implement DetailList in *Windows*? I would like to tackle this.

**Comment by freakboy3742** [comment]
No specific idea here. As with getting icons and widgets on List/Table view, I suspect the approach will be to find a third-party  open source implementation in C#, and try to duplicate that implementation in Python.

**Comment by t-arn** [comment]
Couldn't we just implement DetailedList as a Table with 1 column and no headers?

**Comment by freakboy3742** [comment]
Table might provide the bones for an implementation; but the catch is that the layout *inside* each cell isn't a simple "icon plus one line of text". Think of a Twitter timeline view.

**Comment by saroad2** [comment]
I believe this was fixed in #1574
😄
